presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -402,12 +403,16 @@ private ObjectMetadata getS3ObjectMetadata(Path path)
                             STATS.newMetadataCall();
                             return s3.getObjectMetadata(uri.getHost(), keyFromPath(path));
                         }
-                        catch (AmazonS3Exception e) {
-                            if (e.getStatusCode() == SC_NOT_FOUND) {
-                                return null;
-                            }
-                            else if (e.getStatusCode() == SC_FORBIDDEN) {
-                                throw new UnrecoverableS3OperationException(e);
+                        catch (Exception e) {
```

Comment:
This should be `RuntimeException`, since the block can't throw checked exceptions